packages/main/src/components/AnalyticalTable/TableBody/VirtualTableBody.tsx

```diff
@@ -37,17 +38,28 @@ export const VirtualTableBody = (props: VirtualTableBodyProps) => {
     tableRef,
     visibleColumnsWidth,
     parentRef,
-    overscanCountHorizontal
+    overscanCountHorizontal,
+    renderRowSubComponent
   } = props;
 
+  const rowSubComponentsHeight = useRef({});
+
   const itemCount = Math.max(minRows, rows.length);
   const overscan = overscanCount ? overscanCount : Math.floor(visibleRows / 2);
 
   const consolidatedParentRef = useConsolidatedRef(parentRef);
   const rowVirtualizer = useVirtual({
     size: itemCount,
     parentRef: consolidatedParentRef,
-    estimateSize: React.useCallback(() => internalRowHeight, [internalRowHeight]),
+    estimateSize: React.useCallback(
+      (index) => {
+        if (renderRowSubComponent && rows[index].isExpanded && rowSubComponentsHeight.current?.[index]) {
+          return internalRowHeight + rowSubComponentsHeight.current?.[index];
```

Comment:
```suggestion
          return internalRowHeight + rowSubComponentsHeight.current?.[index] ?? 0;
```
Otherwise it would be number + undefined = NaN